persistence/src/main/java/fi/otavanopisto/pyramus/dao/base/PersonDAO.java

```diff
@@ -298,6 +295,15 @@ public SearchResult<Person> searchPersonsBasic(int resultsPerPage, int page, Str
       break;
     }
     
+    // #1416 Limit results to study programmes of staff member
```

Comment:
Should we return empty list if none are specified? (Same Q for the change below.)

Reply:
I wouldn't mix any application logic to low-level, "dumb" DAO methods. Same as with, say, organization, which is pretty much mandatory everywhere but as far as this method is concerned, it is only added as a search criteria when not null.